src/chat.ts

```diff
@@ -114,11 +81,17 @@ export class ChatProvider implements vscode.WebviewViewProvider {
         }
       }
     );
+  }
 
-    while (this._unhandledMessages.length > 0) {
-      const message = this._unhandledMessages.shift();
-      this.addChatResult(message);
+  public postCommand(command: any) {
+    // Intercept the addRecipes command to make sure they're added to the UI
+    switch (command.command) {
+      case "recipes/addRecipes": {
+        this.recipes = command.recipes;
+        break;
+      }
     }
+    this._view.webview.postMessage(command);
```

Comment:
<!-- Questions are appropriate if you have a potential concern but are not quite sure if it's relevant or not. Asking the author for clarification or investigation can lead to a quick resolution.  -->
**❔ question:** ‏Why is this a special case?

Is this for `Ask Sourcery`? If so please update the comment to make that a little clearer